src/components/Layout/Layout.tsx

```diff
@@ -118,26 +118,15 @@ export const Layout = ({
   const shouldShowGen2Banner = GEN2BANNER_URLS.includes(asPathWithNoHash);
 
   let currentPlatform = DEFAULT_PLATFORM;
-  const homepageNode = directory as PageNode;
-  let rootMenuNode;
 
   const isGen2 = asPathWithNoHash.split('/')[1] === 'gen2';
   const isContributor = asPathWithNoHash.split('/')[1] === 'contribute';
   const currentGlobalNavMenuItem = isContributor ? 'Contribute' : 'Docs';
   const isPrev = asPathWithNoHash.split('/')[2] === 'prev';
-  const searchParam = isGen2 ? 'gen2' : '[platform]';
   const showNextPrev = NEXT_PREVIOUS_SECTIONS.some((section) => {
     return asPathWithNoHash.includes(section) && !asPathWithNoHash.endsWith(section);
   })
 
-  if (homepageNode?.children && homepageNode.children.length > 0) {
```

Comment:
Moving all of this logic into the menu component, now that we are up to 5 different menus it feels too cluttered to keep all of that in Layout.